### PR TITLE
[FEATURE] Make Authelia serve over TLS in all suites

### PR DESCRIPTION
**Pull request by clems4ever**
In order to simplify the configuration of suites, we decided to choose between using TLS or not in Authliea. TLS has been chosen because it involves configuration that could be useful for people and that must be covered with tests anyway.

### CODE REVIEW

File: internal/suites/example/compose/authelia/docker-compose.backend.dev.yml [hunk 1]
```diff
@@ -14,10 +14,13 @@ services:
       - '../..:/app'
       - '${GOPATH}:/go'
     labels:
+      # Traefik 1.x
       - 'traefik.frontend.rule=Host:login.example.com;PathPrefix:/api'
+      # Traefik 2.x
       - 'traefik.http.routers.authelia_backend.rule=Host(`login.example.com`) && PathPrefix(`/api`)'
       - 'traefik.http.routers.authelia_backend.entrypoints=https'
       - 'traefik.http.routers.authelia_backend.tls=true'
+      - 'traefik.http.services.authelia-backend-authelia.loadbalancer.server.scheme=https'
```
Comment (nightah) [inline internal/suites/example/compose/authelia/docker-compose.backend.dev.yml:23]:
```suggestion
      - 'traefik.http.services.authelia_backend.loadbalancer.server.scheme=https'
```

File: internal/suites/example/compose/authelia/docker-compose.backend.dist.yml [hunk 2]
```diff
@@ -3,10 +3,14 @@ services:
...
-      - 'traefik.http.routers.authelia.entrypoints=https'
-      - 'traefik.http.routers.authelia.tls=true'
+      # Traefik 1.x
+      - 'traefik.frontend.rule=Host:login.example.com;PathPrefix:/api'
+      - 'traefik.protocol=https'
+      # Traefik 2.x
+      - 'traefik.http.routers.authelia_backend.rule=Host(`login.example.com`) && PathPrefix(`/api`)'
+      - 'traefik.http.routers.authelia_backend.entrypoints=https'
+      - 'traefik.http.routers.authelia_backend.tls=true'
+      - 'traefik.http.services.authelia-backend-authelia.loadbalancer.server.scheme=https'
```
Comment (nightah) [inline internal/suites/example/compose/authelia/docker-compose.backend.dist.yml:13]:
```suggestion
      - 'traefik.http.services.authelia_backend.loadbalancer.server.scheme=https'
```

File: internal/suites/example/compose/authelia/docker-compose.frontend.dev.yml [hunk 3]
```diff
@@ -14,7 +14,9 @@ services:
       - './example/compose/authelia/resources/:/resources'
       - '../../web:/app'
     labels:
+    # Traefik 1.x
```
Comment (nightah) [inline internal/suites/example/compose/authelia/docker-compose.frontend.dev.yml:17]:
```suggestion
      # Traefik 1.x
```

File: internal/suites/example/compose/authelia/docker-compose.frontend.dist.yml [hunk 4]
```diff
@@ -4,5 +4,15 @@ services:
...
     volumes:
       - './example/compose/authelia/resources/nginx.conf:/etc/nginx/nginx.conf'
+    labels:
+      # Traefik 1.x
+      - 'traefik.frontend.rule=Host:login.example.com'
+      # Traefik 2.x
+      - 'traefik.http.routers.authelia_frontend.rule=Host(`login.example.com`)'
+      - 'traefik.http.routers.authelia_frontend.entrypoints=https'
+      - 'traefik.http.routers.authelia_frontend.tls=true'
+      - 'traefik.http.services.authelia-frontend-authelia.loadbalancer.server.port=3000'
```
Comment (nightah) [inline internal/suites/example/compose/authelia/docker-compose.frontend.dist.yml:14]:
```suggestion
      - 'traefik.http.services.authelia_frontend.loadbalancer.server.port=3000'
```

Comment (nightah) [reply internal/suites/example/compose/authelia/docker-compose.frontend.dist.yml:14]: Actually I don’t believe this is even necessary. Traefik will pick up the port from the expose param so this shouldn’t be required.

Comment (nightah) [reply internal/suites/example/compose/authelia/docker-compose.frontend.dist.yml:14]: On second thought default nginx exposes 80 iirc so you’ll have multiple. Might need to double check if there isn’t a default expose by adding it in the compose that will suffice. If not you will need this.

File: internal/suites/example/compose/haproxy/haproxy.cfg [hunk 5]
```diff
@@ -42,7 +44,17 @@ frontend fe_http
...
+    server authelia-backend authelia-backend:9091 ssl verify none
+
+backend be_auth_request
+  mode http
+  server proxy 127.0.0.1:8085
+
+listen be_auth_request_proxy
+  mode http
+  bind 127.0.0.1:8085
+
```
Comment (nightah) [inline internal/suites/example/compose/haproxy/haproxy.cfg:56]:
```suggestion
```

File: internal/suites/example/compose/authelia/docker-compose.frontend.dist.yml (hunk 4 above)
Comment (clems4ever) [reply internal/suites/example/compose/authelia/docker-compose.frontend.dist.yml:14]: I actually added it because it was not able to pick it up correctly.

**Pull request merged**